llmfoundry/callbacks/hf_checkpointer.py

```diff
@@ -218,6 +253,12 @@ def _is_last_batch(self, state: State):
 
         return False
 
+    def _all_child_processes_done(self) -> bool:
```

Comment:
we should have a timeout on this func in case something goes wrong. rn this upload can't more than hour anyway, so maybe set it to that?

Reply:
sounds good, i'll add to the `elif event == Event.FIT_END:` block